### PR TITLE
ParagraphBuilder should be able to build a paragraph

### PR DESCRIPTION
**Pull request by abarth**
This patch start down the road of implementing text layout and painting without
the DOM. We can construct a basic paragraph consisting of a single run of text
and we can get through layout without crashing.

### CODE REVIEW

File: sky/engine/core/css/CSSFontSelector.cpp [hunk 1]
```diff
@@ -43,14 +43,13 @@ namespace blink {
 
 CSSFontSelector::CSSFontSelector(Document* document)
     : m_document(document)
-    , m_genericFontFamilySettings(document->frame()->settings()->genericFontFamilySettings())
 {
+    if (m_document)
+        m_genericFontFamilySettings = document->frame()->settings()->genericFontFamilySettings();
```
Comment (Hixie) [inline sky/engine/core/css/CSSFontSelector.cpp]: what are generic font family settings and what's our story for getting some without documents?

File: sky/engine/core/css/FontSize.cpp [hunk 2]
```diff
@@ -53,7 +53,7 @@ float FontSize::getComputedSizeFromSpecifiedSize(const Document* document, bool
     // However we always allow the page to set an explicit pixel size that is smaller,
```
Comment (Hixie) [inline sky/engine/core/css/FontSize.cpp]: We should probably drop the minimum font size logic entirely, or hoist it up into Dart. The code here has no idea what transforms are being applied, for one.

File: sky/engine/core/css/FontSize.cpp [hunk 3]
```diff
@@ -53,7 +53,7 @@ float FontSize::getComputedSizeFromSpecifiedSize(const Document* document, bool
...
     // since sites will mis-render otherwise (e.g., http://www.gamespot.com with a 9px minimum).
 
-    Settings* settings = document->settings();
+    Settings* settings = document ? document->settings() : nullptr;
     if (!settings)
```
Comment (Hixie) [inline sky/engine/core/css/FontSize.cpp]: why does this line even exist. The rest of the function doesn't _use_ the settings, as far as I can tell.

**Comment by Hixie** [comment]
LGTM

File: sky/engine/core/css/FontSize.cpp (hunk 3 above)
Comment (abarth) [inline sky/engine/core/css/FontSize.cpp]: We'll probably delete a lot of this code once we cut over to using paragraph builder.

File: sky/engine/core/css/FontSize.cpp (hunk 2 above)
Comment (abarth) [inline sky/engine/core/css/FontSize.cpp]: Yeah, it's nonsense code.

File: sky/engine/core/css/CSSFontSelector.cpp (hunk 1 above)
Comment (abarth) [inline sky/engine/core/css/CSSFontSelector.cpp]: We can actually just delete this line of code if you prefer.  There's no way to modify the genericFontFamilySettings in Settings today anyway.

**Pull request merged**